vald/evm/rpc/arbitrum.go

```diff
@@ -0,0 +1,105 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/axelarnetwork/utils/funcs"
+)
+
+var (
+	nodeInterfaceAddress = common.HexToAddress("0x00000000000000000000000000000000000000C8") // https://github.com/OffchainLabs/nitro/blob/master/contracts/src/node-interface/NodeInterface.sol
+	nodeInterfaceABI     = funcs.Must(abi.JSON(strings.NewReader(
+		`[
+			{
+				"inputs": [
+					{
+						"internalType": "bytes32",
+						"name": "blockHash",
+						"type": "bytes32"
+					}
+				],
+				"name": "getL1Confirmations",
+				"outputs": [
+					{
+						"internalType": "uint64",
+						"name": "confirmations",
+						"type": "uint64"
+					}
+				],
+				"stateMutability": "view",
+				"type": "function"
+			}
+		]`,
+	)))
+	getL1ConfirmationsMethod = funcs.Must(nodeInterfaceABI.MethodById(common.Hex2Bytes("e5ca238c")))
+)
+
+// arbitrumClient implements ArbitrumClient
+type arbitrumClient struct {
+	*ethereumClient
+	l1Client *ethereum2Client
+}
+
+func newArbitrumClient(ethereumClient *ethereumClient, l1Client Client) (*arbitrumClient, error) {
+	// TODO: verify that the given l1 client corresponds to the Arbitrum chain, but how?
+	eth2Client, ok := l1Client.(*ethereum2Client)
+	if !ok {
+		return nil, fmt.Errorf("l1 client has to be ethereum 2.0 for arbitrum")
+	}
```

Comment:
we not change the signature to `*ethereum2Client`, if this can never succeed otherwise?